File: C++/A1/025.cpp
#include <iostream>
using namespace std;
int main(){
    string A;
    cin >> A;
    int len = A.length();
    switch (toupper(A[0]))
    {
    case 'A':
        cout << "ace";
        break;
    case 'J':
        cout << "jack";
        break;
    case 'Q':
        cout << "queen";
        break;
    case 'K':
        cout << "king";
        break;
    default:
        cout << A[0];
        break;
    }
    if(len == 3)cout << A[1];
    cout << " of ";
    switch (toupper(A[len-1]))
    {
    case 'D':
        cout << "diamonds";
        break;
    case 'H':
        cout << "hearts";
        break;
    case 'S':
        cout << "spades";
        break;
    case 'C':
        cout << "clubs";
        break;
    default:
        break;
    }
    
}

File: Python/A1/025.py
Card = input().lower()
rank = Card[:-1]
suit = Card[-1]
ranks = {
    "a": "ace", "j": "jack", "q": "queen", "k": "king",
    "2": "2", "3": "3", "4": "4", "5": "5", 
    "6": "6", "7": "7", "8": "8", "9": "9", "10": "10"
}
suits = {
    "d": "diamonds",
    "h": "hearts",
    "s": "spades",
    "c": "clubs"
}
print(f"{ranks[rank]} of {suits[suit]}")